rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRAData.java

```diff
@@ -97,18 +83,102 @@ public int hashCode() {
+        private String clientId;
+        private LRAStatus status;
+        private boolean isTopLevel;
+        private boolean isRecovering;
+        private long startTime;
+        private long finishTime;
+        private int httpStatus;
+        private String responseData;
+
+        private Builder() {
+            // no public instance
+        }
+
+        public static BuilderWithLraId instanceOf() {
+            return new Builder();
+        }
+
+        public BuilderWithClientId lraId(String lraId) {
+            this.lraId = lraId;
+            return this;
+        }
+        public BuilderWithStatus clientId(String clientId) {
+            this.clientId = clientId;
+            return this;
+        }
+        public BuilderWithTopLevel status(LRAStatus lraStatus) {
+            this.status = lraStatus;
+            return this;
+        }
+        public BuilderWithRecovering topLevel(boolean isTopLevel) {
+            this.isTopLevel = isTopLevel;
+            return this;
+        }
+        public BuilderWithStartTime recovering(boolean isRecovering) {
+            this.isRecovering = isRecovering;
+            return this;
+        }
+        public BuilderWithFinishTime startTime(long startTime) {
+            this.startTime = startTime;
+            return this;
+        }
+        public BuilderWithHttpStatus finishTime(long finishTime) {
+            this.finishTime = finishTime;
+            return this;
+        }
+        public BuilderWithResponseData httpStatus(int httpStatus) {
+            this.httpStatus = httpStatus;
+            return this;
+        }
+        public BuilderFinal responseData(String responseData) {
+            this.responseData = responseData;
+            return this;
+        }
+        public LRAData build() {
+            return new LRAData(this);
+        }
     }
 
+    public static interface BuilderWithLraId {
+        BuilderWithClientId lraId(String lraId);
+    }
+    public static interface BuilderWithClientId {
+        BuilderWithStatus clientId(String clientId);
+    }
+    public static interface BuilderWithStatus {
+        BuilderWithTopLevel status(LRAStatus lraStatus);
+    }
+    public static interface BuilderWithTopLevel {
+        BuilderWithRecovering topLevel(boolean isTopLevel);
+    }
+    public static interface BuilderWithRecovering {
+        BuilderWithStartTime recovering(boolean isRecovering);
+    }
+    public static interface BuilderWithStartTime {
+        BuilderWithFinishTime startTime(long startTime);
+    }
+    public static interface BuilderWithFinishTime {
+        BuilderWithHttpStatus finishTime(long finishTime);
+    }
+    public static interface BuilderWithHttpStatus {
+        BuilderWithResponseData httpStatus(int httpStatus);
+    }
+    public static interface BuilderWithResponseData {
+        BuilderFinal responseData(String responseData);
+    }
+    public static interface BuilderFinal {
+        LRAData build();
+    }
```

Comment:
Sorry, but I think that the Builder here is overcomplicated. What would be the reason to keep the order of methods in the builder? Also since all `BuilderWith*` interfaces have only one method which represents next step in the building process then all parameters are required, and so this is just a really complicated way of writing an all args constructor :)

Reply:
I can remove the `Builder` if you both think it's over-complicated. My position is that the creation of an object with constructor of 8 arguments are pretty messy and builder helps here to put all the values at the right place.